auth_mellon_config.c

```diff
@@ -470,6 +479,78 @@ static const char *am_set_module_config_int_slot(cmd_parms *cmd,
     return ap_set_int_slot(cmd, am_get_mod_cfg(cmd->server), arg);
 }
 
+/* This function handles the MellonDiagnosticsFile configuration directive.
+ * It emits as warning in the log file if Mellon is not built with
+ * diagnostics enabled.
+ *
+ * Parameters:
+ *  cmd_parms *cmd       The command structure for this configuration
+ *                       directive.
+ *  void *struct_ptr     Pointer to the current directory configuration.
+ *                       NULL if we are not in a directory configuration.
+ *  const char *arg      The string argument following this configuration
+ *                       directive in the configuraion file.
+ *
+ * Returns:
+ *  NULL on success or an error string on failure.
+ */
+static const char *am_set_module_diag_file_slot(cmd_parms *cmd,
+                                                    void *struct_ptr,
+                                                    const char *arg)
+{
+#ifdef ENABLE_DIAGNOSTICS
+    return ap_set_file_slot(cmd, am_get_diag_cfg(cmd->server), arg);
+#else
+    ap_log_error(APLOG_MARK, APLOG_WARNING, 0, cmd->server,
+                 "%s has no effect because Mellon was not compiled with"
+                 " diagnostics enabled, use ./configure --enable-diagnostics"
+                 " at build time to turn this feature on.",
+                 cmd->directive->directive);
```

Comment:
Sorry for bringing this up so late, but I just discovered that I am unable to find this log message anywhere. I tried tweaking the log level up to `APLOG_EMERG`, and it still does not show up. I was only able to make it show up by replacing `cmd->server` with `NULL`, at which point it was logged to stderr.

Is it just something weird with my Apache configuration, or is logging during configuration parsing problematic (maybe because logging isn't configured yet?).

A simple alternative may be to turn this into an actual error instead, i.e. `return apr_psprintf("%s is not supported because Mellon was not compiled [...]");`

Reply:
I had tested this but that was before the change over to initializing inside the (virtual) server blocks, that apparently broke the emission of this log message. After running Apache in gdb I was able to figure out why. Your suspicion was essentially correct, the per module log levels had not yet been configured and the server log level was APLOG_UNSET. However the log file is set up and working at this point. There is a special condition in ap_log_core that bypasses the aforementioned checks. If the log level of the message is APLOG_NOTICE the message will be emitted. Apparently APLOG_NOTICE is there to handle exactly these cases, or so it would seem.

This fix is trivial, just change APLOG_WARNING to APLOG_NOTICE. I will push a new version of the PR with the fix.